src/distributed/no_mpi.c

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file distributed/no_mpi.c
+ *
+ * @brief MPI Support Module
+ *
+ * This module implements all basic MPI facilities to let the distributed
+ * execution of a simulation model take place consistently.
+ *
+ * Several facilities are thread-safe, others are not. Check carefully which
+ * of these can be used by worker threads without coordination when relying
+ * on this module.
+ *
+ * SPDX-FileCopyrightText: 2008-2021 HPDCS Group <rootsim@googlegroups.com>
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+#include <distributed/mpi.h>
+
+void mpi_global_init(int *argc_p, char ***argv_p)
+{
+	(void)argc_p;
+	(void)argv_p;
+}
+
+void mpi_global_fini(void) {}
+
+void mpi_remote_msg_send(struct lp_msg *msg, nid_t dest_nid)
+{
+	(void)msg;
+	(void)dest_nid;
+	__builtin_unreachable();
```

Comment:
I am not sure about the `__builtin_unreachable()` here. If I am getting it right, we assume this point shall never be reached because the runtime will never issue a remote message send because all LPs are on local threads.

Then maybe we should replace `__builtin_unreachable()` with `abort()`. The reason is that if the code flow reaches `__builtin_unreachable()`, the execution is not expected to abort, but we enter some undefined behaviour. Conversely, we shouldn't care about the optimisations that the compiler would make thanks to `__builtin_unreachable()` because this function should never be called. While an `abort()` would likely help to spot a bug in the caller functions if something goes wrong.

Am I right in my understanding?

Reply:
Yep, you are right. Consider though, that compilers nowadays are very smart; with LTO, I'm quite confident that code calling this function would get completely optimized away. That wouldn't be the case with an explicit ```abort()``` call. In my opinion, a good compromise would be to put ```assert(0)``` calls, so that building in release mode would enable the compiler to optimize away dead code while maintaining the ability to diagnose misbehaviours  in debug mode. What do you think about it?

Reply:
I totally agree! Let's go with `assert(0)`. Thanks!